File: app/src/main/java/de/sneakpeek/data/MovieQueryResult.java
package de.sneakpeek.data;


import android.os.Parcel;
import android.os.Parcelable;

import com.squareup.moshi.Json;

import java.util.List;

public class MovieQueryResult implements Parcelable {

    public final static Parcelable.Creator<MovieQueryResult> CREATOR = new Creator<MovieQueryResult>() {

        public MovieQueryResult createFromParcel(Parcel in) {
            MovieQueryResult instance = new MovieQueryResult();
            instance.posterPath = ((String) in.readValue((String.class.getClassLoader())));
            instance.adult = ((boolean) in.readValue((boolean.class.getClassLoader())));
            instance.overview = ((String) in.readValue((String.class.getClassLoader())));
            instance.releaseDate = ((String) in.readValue((String.class.getClassLoader())));
            in.readList(instance.genreIds, (java.lang.Integer.class.getClassLoader()));
            instance.id = ((int) in.readValue((int.class.getClassLoader())));
            instance.originalTitle = ((String) in.readValue((String.class.getClassLoader())));
            instance.originalLanguage = ((String) in.readValue((String.class.getClassLoader())));
            instance.title = ((String) in.readValue((String.class.getClassLoader())));
            instance.backdropPath = ((String) in.readValue((String.class.getClassLoader())));
            instance.popularity = ((double) in.readValue((double.class.getClassLoader())));
            instance.voteCount = ((int) in.readValue((int.class.getClassLoader())));
            instance.video = ((boolean) in.readValue((boolean.class.getClassLoader())));
            instance.voteAverage = ((double) in.readValue((double.class.getClassLoader())));
            return instance;
        }

        public MovieQueryResult[] newArray(int size) {
            return (new MovieQueryResult[size]);
        }

    };
    @Json(name = "poster_path") public String posterPath;
    @Json(name = "adult") public boolean adult;
    @Json(name = "overview") public String overview;
    @Json(name = "release_date") public String releaseDate;
    @Json(name = "genre_ids") public List<Integer> genreIds = null;
    @Json(name = "id") public int id;
    @Json(name = "original_title") public String originalTitle;
    @Json(name = "original_language") public String originalLanguage;
    @Json(name = "title") public String title;
    @Json(name = "backdrop_path") public String backdropPath;
    @Json(name = "popularity") public double popularity;
    @Json(name = "vote_count") public int voteCount;
    @Json(name = "video") public boolean video;
    @Json(name = "vote_average") public double voteAverage;

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(posterPath);
        dest.writeValue(adult);
        dest.writeValue(overview);
        dest.writeValue(releaseDate);
        dest.writeList(genreIds);
        dest.writeValue(id);
        dest.writeValue(originalTitle);
        dest.writeValue(originalLanguage);
        dest.writeValue(title);
        dest.writeValue(backdropPath);
        dest.writeValue(popularity);
        dest.writeValue(voteCount);
        dest.writeValue(video);
        dest.writeValue(voteAverage);
    }

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/de/sneakpeek/data/MovieQueryResult.kt
package de.sneakpeek.data


import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import de.sneakpeek.util.createParcel

class MovieQueryResult(@Expose @SerializedName("poster_path") var posterPath: String,
                       @Expose @SerializedName("adult") var adult: Boolean = false,
                       @Expose @SerializedName("overview") var overview: String,
                       @Expose @SerializedName("release_date") var releaseDate: String,
                       @Expose @SerializedName("id") var id: Int = 0,
                       @Expose @SerializedName("original_title") var originalTitle: String,
                       @Expose @SerializedName("original_language") var originalLanguage: String,
                       @Expose @SerializedName("title") var title: String,
                       @Expose @SerializedName("backdrop_path") var backdropPath: String,
                       @Expose @SerializedName("popularity") var popularity: Double = 0.toDouble(),
                       @Expose @SerializedName("vote_count") var voteCount: Int = 0,
                       @Expose @SerializedName("video") var video: Boolean = false,
                       @Expose @SerializedName("vote_average") var voteAverage: Double = 0.toDouble()) : Parcelable {

    private constructor(source: Parcel) : this(
            source.readString(),
            1 == source.readInt(),
            source.readString(),
            source.readString(),
            source.readInt(),
            source.readString(),
            source.readString(),
            source.readString(),
            source.readString(),
            source.readDouble(),
            source.readInt(),
            1 == source.readInt(),
            source.readDouble()
    )

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(posterPath)
        dest.writeValue(adult)
        dest.writeValue(overview)
        dest.writeValue(releaseDate)
        dest.writeValue(id)
        dest.writeValue(originalTitle)
        dest.writeValue(originalLanguage)
        dest.writeValue(title)
        dest.writeValue(backdropPath)
        dest.writeValue(popularity)
        dest.writeValue(voteCount)
        dest.writeValue(video)
        dest.writeValue(voteAverage)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField @Suppress("unused")
        var CREATOR = createParcel { MovieQueryResult(it) }
    }
}